mailpoet/tests/integration/API/JSON/v1/AutomatedLatestContentTest.php

```diff
@@ -43,4 +43,45 @@ public function testItGetTerms() {
     $this->assertCount(1, $response->data);
     $this->assertSame('Uncategorized', $response->data['0']->name);
   }
+
+  public function testItGetsTransformedPostsWithDifferentStatus() {
+    $currentUserId = wp_get_current_user()->ID;
+    wp_set_current_user(1);
+
+    $stati = ['future', 'draft', 'publish', 'pending', 'private'];
+    $types = ['posts', 'products'];
+    foreach ($types as $type) {
+      foreach ($stati as $status) {
```

Comment:
Is there a reason you generate all the options in the test directly instead of using a `dataProvider`? As is the test will stop as soon as a single failure is encountered, right? Maybe that's desired.

Reply:
Thanks for pointing that out. That was negligent. Improved in 49429b219